File: app/src/main/java/maderski/bluetoothautoplaymusic/Controls/WifiControl.java
package maderski.bluetoothautoplaymusic.Controls;

import android.content.Context;
import android.net.wifi.WifiManager;
import android.widget.Toast;

/**
 * Created by Jason on 2/26/17.
 */

public class WifiControl {

    public static void wifiON(Context context, boolean enable) {
        WifiManager wifiManager = (WifiManager) context.getApplicationContext().getSystemService(Context.WIFI_SERVICE);
        wifiManager.setWifiEnabled(enable);
        String toastMessage = enable ? "Wifi turned ON" : "Wifi turned OFF";
        Toast.makeText(context, toastMessage, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/maderski/bluetoothautoplaymusic/controls/WifiControl.kt
package maderski.bluetoothautoplaymusic.controls

import maderski.bluetoothautoplaymusic.helpers.ToastHelper
import maderski.bluetoothautoplaymusic.wrappers.StringResourceWrapper
import maderski.bluetoothautoplaymusic.wrappers.SystemServicesWrapper

/**
 * Created by Jason on 2/26/17.
 */

class WifiControl(
        private val systemServicesWrapper: SystemServicesWrapper,
        private val stringResourceWrapper: StringResourceWrapper,
        private val toastHelper: ToastHelper
) {
    fun wifiON(enable: Boolean) {
        val wifiManager = systemServicesWrapper.wifiManager
        wifiManager.isWifiEnabled = enable
        val toastMessage = if (enable) stringResourceWrapper.wifiTurnedOn else stringResourceWrapper.wifiTurnedOff
        toastHelper.displayMessage(toastMessage)
    }

    fun isWifiON(): Boolean {
        val wifiManager = systemServicesWrapper.wifiManager
        return wifiManager.isWifiEnabled
    }
}
